credd.go

```diff
@@ -0,0 +1,287 @@
+package htcondor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/PelicanPlatform/classad/classad"
+)
+
+// CredType enumerates the credential types supported by the credd.
+// Mirrors the Python bindings CredType enum (Password, Kerberos, OAuth).
+type CredType string
+
+const (
+	// CredTypePassword stores legacy password credentials.
+	CredTypePassword CredType = "Password"
+	// CredTypeKerberos stores Kerberos credentials.
+	CredTypeKerberos CredType = "Kerberos"
+	// CredTypeOAuth stores OAuth2 credentials.
+	CredTypeOAuth CredType = "OAuth"
+)
+
+// ErrCredentialNotFound is returned when the requested credential does not exist.
+var ErrCredentialNotFound = errors.New("credential not found")
+
+// CredentialStatus describes whether a credential exists and when it was last updated.
+type CredentialStatus struct {
+	Exists    bool       `json:"exists"`
+	UpdatedAt *time.Time `json:"updated_at,omitempty"`
+}
+
+// ServiceStatus describes the state of a service credential.
+type ServiceStatus struct {
+	Service   string     `json:"service"`
+	Handle    string     `json:"handle,omitempty"`
+	Exists    bool       `json:"exists"`
+	UpdatedAt *time.Time `json:"updated_at,omitempty"`
+}
+
+// CreddClient defines the operations exposed by the credd (credential daemon).
+// The methods mirror the Python bindings htcondor2.Credd API.
+type CreddClient interface {
+	AddPassword(ctx context.Context, password string, user string) error
```

Comment:
Looks like the "Password" type is Windows-specific.  Let's not add it until someone asks for it.